app/src/interfaces/controllers/controller.go

```diff
@@ -37,3 +38,118 @@ func (c Controller) Index(ctx echo.Context) error {
 	}
 	return ctx.Render(http.StatusOK, "article_list.html", articles)
 }
+
+func (c Controller) AllTodos(ctx echo.Context) error {
+	todos, err := c.Interactor.GetUndeletedTodos()
+	if err != nil {
+		log.Print(err)
+		return ctx.Render(http.StatusInternalServerError, "500.html", nil)
+	}
+
+	return ctx.Render(http.StatusOK, "all_todos.html", todos)
+}
+
+func (c Controller) Detail(ctx echo.Context) error {
+	todoId, strconvErr := c.convertTodoIdToUint(ctx.QueryParam("todo_id"))
+	if strconvErr != nil {
+		log.Print(strconvErr)
+		return ctx.Render(http.StatusInternalServerError, "500.html", nil)
+	}
+
+	todo, err := c.Interactor.GetTodo(todoId)
+	if err != nil {
+		log.Print(err)
+		return ctx.Render(http.StatusInternalServerError, "500.html", nil)
+	}
+
+	return ctx.Render(http.StatusOK, "detail.html", todo)
+}
+
+func (c Controller) NewTodo(ctx echo.Context) error {
+	return ctx.Render(http.StatusOK, "new_todo.html", nil)
+}
+
+func (c Controller) NewTodoSubmit(ctx echo.Context) error {
+	title := ctx.FormValue("title")
+	content := ctx.FormValue("content")
+
+	validationError := validation.TodoValidate(title, content)
+	if validationError.Title != nil || validationError.Content != nil {
+		return ctx.Redirect(http.StatusFound, "/new_todo")
+	}
+
+	err := c.Interactor.InsertNewTodo(title, content)
+	if err != nil {
+		log.Print(err)
+		return ctx.Render(http.StatusInternalServerError, "500.html", nil)
+	}
+
+	return ctx.Redirect(http.StatusFound, "/all_todos")
+}
+
+func (c Controller) EditTodo(ctx echo.Context) error {
+	todoId, strconvErr := c.convertTodoIdToUint(ctx.QueryParam("todo_id"))
+	if strconvErr != nil {
+		log.Print(strconvErr)
```

Comment:
This error variable can also be simply named `err`.🙇‍♂️

Reply:
Thank you very much, I'll fix!